### ISSUE TITLE
When random range being wide, solver give only Min value?

### ISSUE THREAD

**Issue by ollie11** [google/or-tools#939]
My intention is to have random value between -10^6~10^6, however I get only -10^6 with below sample. Anyone shed some light please?
--
from ortools.constraint_solver import pywrapcp
import time
solver = pywrapcp.Solver("graph1")
a =solver.IntVar(-1000000,1000000,'a')
db = solver.Phase([a],solver.CHOOSE_RANDOM, solver.ASSIGN_RANDOM_VALUE)
solver.NewSearch(db)
solver.ReSeed(int(time.time()))
solver.NextSolution()
print a

**Comment by lperron** [comment]
Yes, creating a hole in a large domain will create a large bitset.
Thus we heuristically decide not to do it.

Anyway, you should not use the orignal CP solver but the new CP-SAT solver.

see https://developers.google.com/optimization/cp/cp_solver#cp-sat_example

and

https://github.com/google/or-tools/blob/master/ortools/sat/doc/index.md
Laurent Perron | Operations Research | lperron@google.com | (33) 1 42 68 53
00
